opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -213,10 +213,22 @@ class EclMaterialLawManager
         return *materialLawParams_[elemIdx];
     }
 
-    std::shared_ptr<MaterialLawParams>& materialLawParamsPointerReferenceHack(unsigned elemIdx)
+    MaterialLawParams& materialLawParamsWellConnection(unsigned elemIdx)
     {
         assert(0 <= elemIdx && elemIdx <  materialLawParams_.size());
-        return materialLawParams_[elemIdx];
+        return *materialLawParams_[elemIdx];
+    }
+
+    const MaterialLawParams& materialLawParamsWellConnection(unsigned int connectionIdx) const
```

Comment:
good question. I guess these must come from the completions of the opm-parser Schedule object when the parameter objects are created. If the completions can change during the simulation, the parameter creation probably needs to be done by a public `updateCompletionParams(schedule, reportStepIdx)`  (or so) method. this method will need to be called by the simulator before each report step. better ideas are appreciated...